app/components/tag_list_input_component/tag_list_input_controller.js

```diff
@@ -5,13 +5,23 @@ export default class extends Controller {
   static values = { highlightClass: String };
 
   addTag() {
+    // Check if tag already exist
+    const newTagName = this.newTagTarget.value
+    const tags = this.tagListTarget.value.split(",");
+    const index = tags.indexOf(newTagName);
+    if (index != -1) {
+      // highlight the value in red
+      this.newTagTarget.classList.add("tag-error") 
+      return
+    } 
+    
```

Comment:
I wonder if we should use a Set when assigning new tags and ignore duplicates. Usually we validate in Rails models and a duplicate tag doesn't sound that serious.

Reply:
`act_as_taggable` ignores duplicate, this mostly for UX. It's the same behaviour as [NgTagsInput](https://mbenford.github.io/ngTagsInput/) which is used for tagging on AngularJS pages.